roottest/root/io/xml/CMakeLists.txt

```
@@ -1,9 +1,37 @@
-#-------------------------------------------------------------------------------
-#
-# Placeholder file to translate the tests to the new CTest system. Meanwhile we
-# define a CTest test that calls 'make' in ${CMAKE_CURRENT_SOURCE_DIR}
-#
-#-------------------------------------------------------------------------------
-if (ROOT_xml_FOUND)
-ROOTTEST_ADD_OLDTEST()
-endif()
+
+if (xml)
+  ROOTTEST_ADD_TEST(SaxParserSimple
+                    MACRO execSaxParserSimple.C
+                    WORKING_DIR ${CMAKE_CURRENT_SOURCE_DIR}
+                    OUTREF execSaxParserSimple.ref)
+endif(xml)
+
+ROOTTEST_ADD_TEST(PolyMarker
+                  MACRO runPolyMarker.C
+                  OUTREF PolyMarker.ref)
+
+ROOTTEST_ADD_TEST(enginexml
+                  MACRO runenginexml.C
+                  WORKING_DIR ${CMAKE_CURRENT_SOURCE_DIR}
+                  OUTREF enginexml.ref)
+
+ROOTTEST_ADD_TEST(XmlDir
+                  MACRO runXmlDir.C
+                  OUTREF XmlDir.ref)
+
+ROOTTEST_GENERATE_DICTIONARY(libXmlTestClasses
```

```suggestion
ROOTTEST_GENERATE_DICTIONARY(libXmlTestDictionaries
```
Part of my (slight) original confusion is  that library name is a slight misnomer (it does not contain any of the implementation of the TestClasses, just their dictionaries).